api/mappers.go

```diff
@@ -181,19 +181,49 @@ func NewPatientDto(patient *patients.Patient) Patient {
 		UpdatedTime:   &patient.UpdatedTime,
 		Summary:       NewSummaryDto(patient.Summary),
 		Reviews:       NewReviewsDto(patient.Reviews),
+		ConnectionRequests: &ProviderConnectionRequests{
+			Abbott: NewConnectionRequestDTO(patient.ProviderConnectionRequests, Abbott),
+			Dexcom: NewConnectionRequestDTO(patient.ProviderConnectionRequests, Dexcom),
+			Twiist: NewConnectionRequestDTO(patient.ProviderConnectionRequests, Twiist),
+		},
 	}
 	if patient.BirthDate != nil && strtodatep(patient.BirthDate) != nil {
 		dto.BirthDate = *strtodatep(patient.BirthDate)
 	}
 	if !patient.LastUploadReminderTime.IsZero() {
 		dto.LastUploadReminderTime = &patient.LastUploadReminderTime
 	}
-	if !patient.LastRequestedDexcomConnectTime.IsZero() {
-		dto.LastRequestedDexcomConnectTime = &patient.LastRequestedDexcomConnectTime
+
+	// Populate the new connection requests structure from the now deprecated lastRequestedDexcomConnectTime
+	if len(dto.ConnectionRequests.Dexcom) == 0 && !patient.LastRequestedDexcomConnectTime.IsZero() {
+		dto.ConnectionRequests.Dexcom = []ProviderConnectionRequest{{
+			ProviderName: Dexcom,
+			CreatedTime: patient.LastRequestedDexcomConnectTime,
+		}}
 	}
+
 	return dto
 }
 
+func NewConnectionRequestDTO(requests patients.ProviderConnectionRequests, provider ProviderId) []ProviderConnectionRequest {
+	empty := make([]ProviderConnectionRequest, 0)
+	if requests == nil {
+		return empty
+	}
+	requestsForProvider, exist := requests[string(provider)]
+	if !exist || len(requests) == 0 {
```

Comment:
Is the intent here to check `requests` to be empty or `requestsForProvider`

Reply:
The intent was to check `requestsForProvider` so the logic was incorrect, good catch. I have simplified this.